runtime/commonTest/src/kotlinx/serialization/SerialDescriptorBuilderTest.kt

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2017-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization
+
+import kotlinx.serialization.internal.*
+import kotlin.test.*
+
+@ImplicitReflectionSerializer
+class SerialDescriptorBuilderTest {
+
+    @Serializable
+    @SerialName("Wrapper")
+    class Wrapper(val i: Int)
+
+    private val wrapperDescriptor = SerialDescriptor("Wrapper", 1, StructureKind.CLASS) {
+        element("i", IntSerializer.descriptor)
+    }
+
+    private val dataHolderDescriptor = SerialDescriptor("DataHolder", 5, StructureKind.CLASS) {
+        element<String>("string")
+        element("nullableWrapper", wrapperDescriptor.nullable)
+        element("wrapper", wrapperDescriptor)
+        element<Int>("int")
+        element<Long?>("nullableOptionalLong", isOptional = true)
+    }
+
+    @Serializable
+    @SerialName("DataHolder")
+    class DataHolder(
+        val string: String,
+        val nullableWrapper: Wrapper?,
+        val wrapper: Wrapper,
+        val int: Int,
+        val nullableOptionalLong: Long? = null
+    )
+
+    @Test
+    fun testTrivialDescriptor() {
+        Wrapper.serializer().descriptor.assertDescriptorEqualsTo(wrapperDescriptor)
+    }
+
+
+    @Test
+    fun testNestedDescriptors() {
+        DataHolder.serializer().descriptor.assertDescriptorEqualsTo(dataHolderDescriptor)
+    }
+
+    @Serializable
+    @SerialName("Box")
+    class Box<T>(val value: T, val list: List<T>)
+
+    class CustomBoxSerializer<T>(val typeSerializer: KSerializer<T>) {
+        val descriptor: SerialDescriptor = SerialDescriptor("Box", 2, StructureKind.CLASS) {
+            element("value", typeSerializer.descriptor)
+            element("list", ArrayListSerializer(typeSerializer).descriptor)
```

Comment:
use `listSerializer` function?

Reply:
This test was written before this function :)